src/main/java/org/opentripplanner/graph_builder/linking/SimpleStreetSplitter.java

```diff
@@ -408,6 +412,22 @@ else if (ll.getSegmentIndex() == orig.getNumPoints() - 2 && ll.getSegmentFractio
             SplitterVertex v0 = split(edge, ll, temporaryVertex != null, endVertex);
             makeLinkEdges(tstop, v0);
 
+            if (OTPFeature.FlexRouting.isOn() && graph.index != null) {
+                Point p = GeometryUtils
+                    .getGeometryFactory()
+                    .createPoint(v0.getCoordinate());
+                Envelope env = p.getEnvelopeInternal();
+                for (FlexStopLocation flexStopLocation : graph.index.getFlexIndex().locationIndex.query(env)) {
+                    if (edge.canTraverse(new TraverseModeSet(TraverseMode.CAR))
```

Comment:
Use the `isEligibleForCarPickupDropOff()`  here instead.

Reply:
Using `StreetTraversalPermission.PEDESTRIAN_AND_CAR`, as `isEligibleForCarPickupDropOff()` operates on vertices